api/datasets.go

```diff
@@ -227,9 +227,19 @@ func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
 	res := []dsref.VersionInfo{}
 	var err error
 	if params.Raw {
-		resRaw, err = h.ListRawRefs(r.Context(), params)
+		got, _, err := h.inst.Dispatch(r.Context(), "dataset.listrawrefs", params)
+		ok := false
+		resRaw, ok = got.(string)
+		if err != nil || !ok {
+			util.RespondWithError(w, err)
```

Comment:
Calls to `RespondWithError`, same as `WriteErrResponse`, should be followed by a return. We don't want to do anything else in a handler after sending a response.